liquidctl/driver/nzxt_smart_device.py

```diff
@@ -121,39 +159,55 @@
 _MAX_DUTY = 100
 _READ_ENDPOINT = 0x81
 _READ_LENGTH = 21
+_READ_LENGTH_V2 = 60     # for Smart Device V2
 _WRITE_ENDPOINT = 0x1
 _WRITE_LENGTH = 65
 
-
 class NzxtSmartDeviceDriver(UsbHidDriver):
     """USB driver for the NZXT Smart Device and Grid+ V3."""
 
+    DEVICE_SMARTDEV_V1 = 'Smart Device V1'
+    DEVICE_GRID_V3  = 'Grid V3'
+    DEVICE_SMARTDEV_V2 = 'Smart Device V2'
+
     SUPPORTED_DEVICES = [
         (0x1e71, 0x1714, None, 'NZXT Smart Device', {
             'speed_channel_count': 3,
-            'color_channel_count': 1
+            'color_channel_count': 1,
+            'device_type': DEVICE_SMARTDEV_V1
         }),
         (0x1e71, 0x1711, None, 'NZXT Grid+ V3 (experimental)', {
             'speed_channel_count': 6,
-            'color_channel_count': 0
+            'color_channel_count': 0,
+            'device_type': DEVICE_GRID_V3
+        }),
+        (0x1e71, 0x2006, None, 'NZXT Smart Device V2 (experimental)', {
+            'speed_channel_count': 3,
+            'color_channel_count': 2,
+            'device_type': DEVICE_SMARTDEV_V2
```

Comment:
Agreed. Let's wait until other code review issues are resolved.

Reply:
I just split the driver into different classes.  Please test and make sure I didn't break the support for V2.

**Note: you'll need the changes in cli.py as well.  It's probably best if you install this branch directly, or at least use a tarball of it (instead of manually copying files).**

Reply:
Very nice. Will download the entire tarball and test the changes.

Reply:
The refactored driver works well!

Had to change Line 314 in order to define keys "led1" and "led2" instead of "led0" and "led1". The code change has been committed. Line 314 went from:
```Python
        color_channels = {'led{}'.format(i + 0): (i)
```
to this:
```Python
        color_channels = {'led{}'.format(i + 1): (i)
```
The `status` function now returns well-aligned columns:
```
Device 1, NZXT Smart Device V2
Fan 2 duty                         40  %
Fan 2 speed                       873  rpm
Firmware version                1.5.0  
LED 1 accessory 1     HUE 2 LED Strip  
LED 1 accessory 2     HUE 2 Underglow  
LED 1 accessory 3     HUE 2 Underglow  
LED 2 accessory 1    AER RGB 2 140 mm  
LED 2 accessory 2    AER RGB 2 140 mm  
LED 2 accessory 3    AER RGB 2 140 mm  
LED 2 accessory 4    AER RGB 2 120 mm  
Noise level                        64  dB
```

Reply:
Ops, no idea how I messed up led[12].

Thanks for fixing it.